src/fsharp/CheckExpressions.fs

```diff
@@ -9770,7 +9770,14 @@ and TcLetBinding cenv isUse env containerInfo declKind tpenv (synBinds, synBinds
 
                 let tmp, _ = mkCompGenLocal m "patternInput" (generalizedTypars +-> tauTy)
 
-                if isUse || isFixed then
+                if isUse then
+                    let isDiscarded = match checkedPat with TPat_wild _ -> true | _ -> false
```

Comment:
This is an unnecessary local

Reply:
I think it adds clarity to what is going on.

Reply:
ok